File: Olympiad/COCI/official/2007/contest6_solutions/prase.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 6 - Task PRASE
  Programming language Pascal
*)

program prase;

var
   i, j, n   : longint;
   red       : array[1..100] of string;
   dosad, rj : longint;

begin
   readln(n);
   rj := 0;
   for i:=1 to n do begin
      readln(red[i]);
      dosad := 0;
      for j:=1 to i-1 do begin
         if red[i] = red[j] then
            dosad := dosad+1;
      end;
      if dosad > i-1-dosad then
         rj := rj+1;
   end;

   writeln(rj);
end.

File: Olympiad/COCI/official/2007/contest6_solutions/prase.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 6 - Task PRASE
  Programming language C++
*/

#include <iostream>
#include <string>
using namespace std;

#define MAXN 100

int main()
{
   int n;
   static string red[MAXN];

   cin >> n;
   int ret = 0;
   for ( int i=0; i<n; ++i ) {
      cin >> red[i];

      int dosad = 0;
      for ( int j=0; j<i; ++j )
         dosad += red[i] == red[j];
      if ( dosad > i-dosad )
         ++ret;                 // prase!
   }

   printf( "%d\n", ret );

   return 0;
}
